src/Workspaces/CSharp/Portable/Extensions/SyntaxTriviaListExtensions.cs

```diff
@@ -47,6 +50,37 @@ public static IEnumerable<SyntaxTrivia> SkipInitialWhitespace(this SyntaxTriviaL
             return triviaList.SkipWhile(t => t.Kind() == SyntaxKind.WhitespaceTrivia);
         }
 
+        private static ImmutableArray<ImmutableArray<SyntaxTrivia>> BreakIntoLines(this SyntaxTriviaList triviaList)
+        {
+            var result = ArrayBuilder<ImmutableArray<SyntaxTrivia>>.GetInstance();
+            var currentLine = ArrayBuilder<SyntaxTrivia>.GetInstance();
+            foreach (var trivia in triviaList)
+            {
+                currentLine.Add(trivia);
+                if (trivia.Kind() == SyntaxKind.EndOfLineTrivia)
+                {
+                    result.Add(currentLine.ToImmutableAndFree());
+                    currentLine = ArrayBuilder<SyntaxTrivia>.GetInstance();
+                }
+            }
+
+            if (currentLine.Count > 0)
+            {
+                result.Add(currentLine.ToImmutableAndFree());
+            }
+
+            return result.ToImmutableAndFree();
+        }
+
+        public static SyntaxTriviaList WithoutLeadingBlankLines(this SyntaxTriviaList triviaList)
+        {
+            return new SyntaxTriviaList(triviaList.BreakIntoLines()
+                .SkipWhile(l => l.All(t =>
```

Comment:
Is it always the case that a blank line will be either exactly end of line trivia, or a single whitespace trivia followed by an end of line trivia? If so I'm wondering if it's a bit easier just to do something (roughly, I'm ignoring some bounds checking):

```
int start = 0;
SyntaxTriviaList list;

while (true)
{
    if (list[start].Kind() == SyntaxKind.EndOfLineTrivia)
        start++;
    else if (list[start].Kind == SyntaxKind.WhitespaceTrivia && list[start + 1].Kind == SyntaxKind.WhitespaceTrivia)
        start += 2;
    else
        break;
}

return new SyntaxTriviaList(list.Skip(start));
```

My worry is in the case where there's a long comment spanning many lines this is fairly expensive to ultimately discover it's a no-op.

Reply:
I changed this